std/math.d

```diff
@@ -4103,9 +4103,55 @@ long lrint(real x) @trusted pure nothrow @nogc
 
             return sign ? -result : result;
         }
+        else static if (F.realFormat == RealFormat.ieeeQuadruple)
+        {
+            long result;
+
+            // Rounding limit when casting from real(128-bit) to ulong.
+            enum real OF = 0x1P+112;
+
+            ulong* vl = cast(ulong*)(&x);
+
+            // Find the exponent and sign
+            ulong msb = vl[MANTISSA_MSB];
+            ulong lsb = vl[MANTISSA_LSB];
+            int exp = cast(int)((msb >> 48) & 0x7fff) - 0x3fff;
+            int sign = cast(int)(msb >> 63);
+            msb &= 0xffff_ffffffff;
+            msb |= 0x10000_00000000;
```

Comment:
The values stored in `msb` and `lsb` by this point are never used. You overwrite them a few lines down without ever reading from them.